rest/route/host_agent.go

```diff
@@ -1254,6 +1262,8 @@ func (h *hostAgentEndTask) Run(ctx context.Context) gimlet.Responder {
 			Message:    fmt.Sprintf("task '%s' not found", h.taskID),
 		})
 	}
+	span.SetAttributes(attribute.Int(evergreen.TaskExecutionOtelAttribute, t.Execution))
```

Comment:
so it means we will only be able to see the attribute on this span right? So I'm assuming these setAttribute function is usually only used for fields that will change throughout the stack. I don't think it matters for this case but it does seem more appropriate to add it to the context since it will never change. Let me know if I'm misunderstanding anything! 

<img width="1328" alt="image" src="https://github.com/evergreen-ci/evergreen/assets/26491602/e831b18a-f18a-4197-bbda-6c8d2c634075">

Reply:
Yes! it will just be on that span.
So if you wanted to facilitate queries on the duration_ms of both `mark-end` spans and `update-blocked-dependencies` grouped by task name you might want to go with `utility.ContextWithAttributes` to set it instead of `span.SetAttributes`.

Reply:
Ahhh, I can see why this would be more useful! Thank you